### DISCUSSION TITLE
Introducing q-otp: Making OTP Input Convenient

### DISCUSSION THREAD

**Discussion by qertis** [quasarframework/quasar#15910] [Quasar App Extensions]
Dear developers,

I am thrilled to introduce a new Quasar extension called q-otp. The purpose of this extension is to provide a convenient and user-friendly OTP (One-Time Password) input for Quasar applications.

We understand that OTP inputs play a crucial role in many applications, especially those that require two-factor authentication or secure verification processes. With q-otp, we aim to simplify the implementation of OTP inputs and enhance the user experience.

We have put a lot of effort into developing q-otp, and we believe it can greatly simplify the implementation of OTP inputs in your Quasar applications. However, we need your help to make it even better.

We invite you to try out q-otp and provide us with your valuable feedback. Whether it's regarding the ease of use, performance, or any suggestions for improvement, your input will be immensely valuable in refining q-otp and making it a valuable tool for the Quasar community.

You can find the q-otp repository on GitHub: [link to repository](https://github.com/qertis/q-otp)

Light theme
![image](https://github.com/quasarframework/quasar/assets/3893228/9e2db540-849b-42b6-9e42-2270d8a61dbb)

Dark theme:
![image](https://github.com/quasarframework/quasar/assets/3893228/7358c71b-e662-4bd6-9255-56b194fcc30a)

Best regards,
Denis Baskovsky

**Comment by rstoenescu** [comment]
Nice! Thanks for sharing with the community.
I'd suggest adding a few screenshots and a demo. This would improve the adoption of your component. Might be a good idea to release it as standalone also (check our "yarn create quasar" > Quasar UI kit) as this will enlarge the audience (devs using our Vite plugin and/or UMD).

**Comment by qertis** [comment]
For developers who use Vite, I can suggest a simpler way to import a plugin, without even initializing the App Extension:
```html
<script lang="ts" setup>
import QOtp from 'quasar-app-extension-q-otp/src/component/QOtp.vue'
</script>
<template>
  ...
  <QOtp />
  ...
</template>
```

Ideally, I would like to have the ability to directly import Quasar components through a CDN. For example, using defineAsyncComponent like this:

```js
app.component(
  "QOtp",
  defineAsyncComponent(
    () =>
      import(
        "https://cdn.jsdelivr.net/npm/quasar-app-extension-q-otp@^0.0.3/src/component/QOtp.vue"
      )
  )
);
```


I tried doing this but found that the functionality of [@vue/compiler-sfc](https://www.npmjs.com/package/@vue/compiler-sfc) doesn't allow for such a mechanism. It seems that it is not capable of rendering TypeScript and SCSS on the fly. I'm eagerly awaiting such a possibility because it would make development easier without relying on the CLI.

**Comment by kamasuPaul** [comment]
Just wondering. Does it support having **6 input boxes instead of 4**

**Reply by yusufkandemir** [reply]
After a very quick look at the repo, it is possible through the `num` prop.
https://github.com/qertis/q-otp/blob/master/src/component/QOtp.vue

**Comment by dilungasr** [comment]
Is it capable of pasting?

**Reply by qertis** [reply]
Yes. Ctrl+V and PasteEvent

**Comment by qertis** [comment]
## New version - 0.1.0

### New design

<img width="219" alt="image" src="https://github.com/quasarframework/quasar/assets/3893228/caf3dd18-4854-4257-9f96-4057cfb15551">

### Validation
<img width="199" alt="image" src="https://github.com/quasarframework/quasar/assets/3893228/07568848-35bc-4571-8483-dfba09390ec9">

### Hint and loading props
<img width="232" alt="image" src="https://github.com/quasarframework/quasar/assets/3893228/595ebd44-0bdb-477d-b5f7-a25a9a35020b">

**Comment by KenKenHipolito** [comment]
Is this responsive out of the box? Currently having a problem the input are not changing width 

![image](https://github.com/quasarframework/quasar/assets/118034758/dc21e0fd-0ecf-4a46-a191-6a1880d04424)

This my current config
![image](https://github.com/quasarframework/quasar/assets/118034758/9abcf459-2acb-4542-a208-e2000dbfc3c4)
